apps/rush-lib/src/logic/installManager/WorkspaceInstallManager.ts

```diff
@@ -587,9 +587,14 @@ export class WorkspaceInstallManager extends BaseInstallManager {
       args.push('--recursive');
       args.push('--link-workspace-packages', 'false');
 
-      // "<package>..." selects the specified package and all direct and indirect dependencies
-      for (const toProject of this.options.toProjects) {
-        args.push('--filter', `${toProject.packageName}...`);
+      const filteredProjects: string[] = this.options.toProjects
+        // "<package>..." selects the specified package and all direct and indirect dependencies
+        .map((p) => p.packageName + '...')
+        // ..."<package>" selects the specified package and all direct and indirect dependents of that package
+        .concat(this.options.fromProjects.map((p) => '...' + p.packageName));
```

Comment:
So, the original PR that brought this functionality in actually had this. I was talked out of it by @octogonz (though I think it may have been a Gitter discussion since I can't find it now...). Main reason it was removed was that I couldn't think of a great justification for how this would be used. So he's likely going to need some convincing :)

Reply:
Our justification is that in order to use the `rush build --from` flag, we need to also install the deps of those dependents coming up from the `--from` flag. If package A relies on B, we can't build A when B changes, if we don't install the dependencies for A via the `--from` flag.

Reply:
Regardless of whether or not upstream projects need to be built, there are plenty of instances where merely referencing the upstream project results in references to the `node_modules` of the upstream project, so it would not surprise me if, unlike in the `rush build` case (where the developer may be running it after an earlier `rush build`), `rush install --from` is completely unusable if it skips the dependencies of upstream local projects.